### DISCUSSION TITLE
Is there a way to add labels to links on the place where I dbclick?

### DISCUSSION THREAD

**Discussion by Naveena1321** [clientIO/joint#2788] [Q&A] [answered]
### Introduction

Have a require to add labels in links/elements where i doubleclicked and select it for text editor by default

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
```ts
paper.on('link:pointerdblclick', (linkView, evt, x, y) => {
   const ratio = linkView.getClosestPointRatio({ x, y });
   linkView.model.appendLabel({ attrs: { text: { text: 'Label' }}, position: { distance: ratio }});
   // update the linkView immediately
   paper.requireView(link);   
   
   const labelTextNode = linkView.findNodeLabel(0, 'text');
   // edit the `labelTextNode` with the TextEditor
});
```

**Reply by Naveena1321** [reply]
Thanks that's very helpful

**Reply by Naveena1321** [reply]
Won't the vertices tools cause issue ?

**Reply by kumilingus** [reply]
It is an issue if the vertex is meant to be created with a single click.
If you need both vertices and labels, maybe you can add a tool button to add labels (see this [example](https://github.com/clientIO/joint/tree/master/examples/link-labels-ts)).

Before click:
<img width="474" alt="before click" src="https://github.com/user-attachments/assets/80be39d7-91b5-4c28-87a0-8a5fac2d8a4e">
After click:
<img width="447" alt="after click" src="https://github.com/user-attachments/assets/179c980b-f074-44e3-9fa2-c83c75007bd9">
